qcelemental/datum.py

```diff
@@ -28,8 +29,35 @@ class Datum(collections.namedtuple('Datum', 'label units data comment doi glossa
 
     """
 
-    def __new__(cls, label, units, data, comment='', doi=None, glossary=''):
-        return super(Datum, cls).__new__(cls, label, units, data, comment, doi, glossary)
+    label: str
+    units: str
+    data: Any
+    comment: str = ''
+    doi: str = None
+    glossary: str = ''
+
+    def __init__(self, label, units, data, *, comment=None, doi=None, glossary=None):
+        kwargs = {'label': label, 'units': units, 'data': data}
+        if comment is not None:
+            kwargs['comment'] = comment
+        if doi is not None:
+            kwargs['doi'] = doi
+        if glossary is not None:
+            kwargs['glossary'] = glossary
+
+        super().__init__(**kwargs)
+
+    @validator('data')
+    def must_be_numerical(cls, v, values, **kwargs):
+        try:
+            ident = 1.0 * v
+        except TypeError:
+            try:
+                ident = Decimal('1.0') * v
+            except TypeError:
+                raise ValueError('Datum data should be float, Decimal, or np.ndarray')
+
+        return v
```

Comment:
As LGTM mentions ident is never used, did you mean to return this?

Reply:
Yes, I had a fix (now pushed). Was waiting to compbine it with any other fixes spotted.